src/core/SampleBuffer.cpp

```diff
@@ -416,7 +412,10 @@ f_cnt_t SampleBuffer::decodeSampleSF( const char * _f,
 	f_cnt_t frames = 0;
 	bool sf_rr = false;
 
-	if( ( snd_file = sf_open( _f, SFM_READ, &sf_info ) ) != NULL )
+
+	QFile f(_f);
```

Comment:
I'm thinking these should all be commented with something along the lines of
```
// Open the file with QFile and pass the descriptor to sf_open_fd instead of directly calling sf_open
// for better filename encoding support. See https://github.com/LMMS/lmms/pull/4401
```
so that people know not to revert this change.

Reply:
Good point.